lms/models/application_settings.py

```diff
@@ -59,6 +62,44 @@ def set(self, group, key, value):
         """
         super().setdefault(group, {})[key] = value
 
+    def set_secret(self, aes_service, group, key, value: str):
+        """
+        Store a setting as a secret.
+
+        Store the setting AES encrypted.
+
+        :param aes_service: AESService to encrypt the string
+        :param group: The name of the group of settings
+        :param key: The key in that group
+        :param value: The value to set
+        """
+        aes_iv = aes_service.build_iv()
+        value = aes_service.encrypt(aes_iv, value)
+
+        # Store both the setting and the IV
+        # We can't store the bytes directly in JSON so we store it as base64
+        super().setdefault(group, {})[key] = base64.b64encode(value).decode("utf-8")
+        super().setdefault(group, {})[f"{key}_aes_iv"] = base64.b64encode(
+            aes_iv
+        ).decode("utf-8")
+
+    def get_secret(self, aes_service, group, key) -> Optional[str]:
```

Comment:
This is not used but  I reckon the PR would be incomplete without the get part.

Reply:
Yep. It would be very strange to be able to encrypt, but not decrypt the values. Also round trip testing would be impossible.